### PR TITLE
*: remove unnecessary metrics

### LINKED ISSUE #6805: The cancel reason of operator should be shown on the monitor
## Enhancement Task
We do have these metrics but we don't add them to the monitor, if we cannot edit the metrics, it's hard to know the actual reason.

### PR DESCRIPTION
**Pull request by rleungx**
### What problem does this PR solve?

Issue Number: Close #6805

### What is changed and how does it work?

Unify the `operatorWaitCounter` and `operatorCounter`.

```commit-message
```

### Check List

Tests

- Unit test

### Release note

```release-note
None.
```

### CODE REVIEW

File: pkg/schedule/operator/operator_controller.go [hunk 1]
```diff
@@ -126,7 +126,7 @@ func (oc *Controller) Dispatch(region *core.RegionInfo, source string, recordOpS
 				recordOpStepWithTTL(op.RegionID())
 			}
 			if oc.RemoveOperator(op, Succeed) {
-				operatorWaitCounter.WithLabelValues(op.Desc(), "promote-success").Inc()
+				operatorCounter.WithLabelValues(op.Desc(), "promote-success").Inc()
```
Comment (JmPotato) [inline pkg/schedule/operator/operator_controller.go:129]: Could we pre-initialize these label values inside the operator struct to reduce the cost here?

**Review by JmPotato — COMMENTED** [review]
The rest LGTM.

File: pkg/schedule/operator/operator_controller.go (hunk 1 above)
Comment (rleungx) [reply pkg/schedule/operator/operator_controller.go:129]: Here we need to use `op.Desc()` which cannot be pre-initialized.

Comment (JmPotato) [reply pkg/schedule/operator/operator_controller.go:129]: Maybe we could initialize it when creating the operator like:

```go
func NewOperator(desc, brief string, regionID uint64, regionEpoch *metapb.RegionEpoch, kind OpKind, approximateSize int64, steps ...OpStep) *Operator {
	return &Operator{
		desc:                  desc,
		promoteSuccessCounter: operatorCounter.WithLabelValues(desc, "promote-success")
	}
}
```

Comment (rleungx) [reply pkg/schedule/operator/operator_controller.go:129]: hmm, what is the difference?

Comment (JmPotato) [reply pkg/schedule/operator/operator_controller.go:129]: Oops, I misunderstood the initialization process here, it's the same if we initialize the metrics each time we create a new operator here.

**Review by JmPotato — APPROVED** [review]

**Review by nolouch — APPROVED** [review]

**Pull request merged**